### PR TITLE
Genericize dpdk

### PR DESCRIPTION
**Pull request by justin-parus**
- Add new and more proper dpdk-testFWD testcase
- Fix old dpdk-testPMD testcase to use multiple flows
- DPDK-TESTCASE-DRIVER and corresponding linux framework now supports arbitrary many VMs
- Replaced useless Prepare-Parameters with Configure-Test. e.g. test case can modify VMs before test start

### CODE REVIEW

File: XML/VMConfigurations/PerformanceTestsConfigurations.xml [hunk 1]
```diff
@@ -36,28 +80,28 @@
...
                     <Protocol>tcp</Protocol>
                     <LocalPort>22</LocalPort>
                     <PublicPort>1112</PublicPort>
                 </EndPoints>
                 <DataDisk></DataDisk>
-                <ExtraNICs>2</ExtraNICs>
+                <ExtraNICs>1</ExtraNICs>
             </VirtualMachine>
         </ResourceGroup>
     </DS15v2_2NIC_x2>
```
Comment (rmsivakanth) [inline XML/VMConfigurations/PerformanceTestsConfigurations.xml]: @justin-parus  Can you please sync your code with master? 
**Your code is 89 commits behind LIS:master.**

I wanted to run it locally and it failed with below error:
```
.\Run-LisaV2.ps1 -EnableTelemetry -TestLocation 'eastus' -RGIdentifier 'sru18' -TestPlatform 'Azure' -StorageAccount 'ExistingStorage_Standard' -UpdateGlobalConfigurationFromSecretsFile -XMLSecretFile '****' -ARMImageName 'Canonical UbuntuServer 18.04-LTS latest' -TestNames 'PERF-DPDK-SINGLE-CORE-PPS-DS4' -CoreCountExceededTimeout '172800' -TestIterations '1' -UseManagedDisk
.\Run-LisaV2.ps1 : The term '.\Run-LisaV2.ps1' is not recognized as the name 
of a cmdlet, function, script file, or operable program. Check the spelling of 
the name, or if a path was included, verify that the path is correct and try 
again.
At line:1 char:1
+ .\Run-LisaV2.ps1 -EnableTelemetry -TestLocation 'eastus' -RGIdentifie ...
+ ~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (.\Run-LisaV2.ps1:String) [], Co 
   mmandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
```

Comment (rmsivakanth) [reply XML/VMConfigurations/PerformanceTestsConfigurations.xml]: @justin-parus , Can you please run all affected tests and put results here.

File: Testscripts/Linux/dpdk-testFWD.sh [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	LogMsg "Ensuring free hugepages"
+	local free_huge_cmd="rm -rf /dev/hugepages/*"
+	for ip in $IP_ADDRS; do
+		ssh ${ip} ${free_huge_cmd}
+	done
+	
+	# start receiver, fowarder in advance so testpmd comes up easily
+	local fwd_recv_duration=$(expr ${test_duration} + 5)
+	
+	local receiver_testfwd_cmd="timeout ${fwd_recv_duration} ${LIS_HOME}/${DPDK_DIR}/build/app/testpmd -l 0-${core} -w ${receiver_busaddr} --vdev='net_vdev_netvsc0,iface=${receiver_iface}' -- --port-topology=chained --nb-cores ${core} --txq ${core} --rxq ${core} --mbcache=512 --txd=4096 --rxd=4096 --forward-mode=rxonly --stats-period 1"
```
[the remark this answers is not in the document]
Comment (justin-parus) [reply Testscripts/Linux/dpdk-testFWD.sh]: fixed

File: Testscripts/Linux/dpdk-testFWD.sh [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+# 	forwaring scenario (i.e. a more correct packet stream that avoids the exception path).
+#	It places testpmd output in $LOG_DIR, and then parses output to calculate avg pps.
+#	The accompanying ps1 script makes sure testpmd performs above the expected threshold.
+#
+#############################################################################
+
+# Requires
+#   - called by install_dpdk in dpdk top level dir
+#   - first argument is install_ip
+function dpdk_configure() {
```
[the remark this answers is not in the document]
Comment (justin-parus) [reply Testscripts/Linux/dpdk-testFWD.sh]: fixed

File: XML/VMConfigurations/PerformanceTestsConfigurations.xml (hunk 1 above)
Comment (justin-parus) [reply XML/VMConfigurations/PerformanceTestsConfigurations.xml]: 10/22/2018 21:00:07 : [INFO ] ~~~~~~~~~~~~~~~TEST END : PERF-DPDK-FWD-PPS-DS15~~~~~~~~~~
10/22/2018 21:00:07 : [INFO ] CURRENT - PASS    - 1
10/22/2018 21:00:07 : [INFO ] CURRENT - FAIL    - 0
10/22/2018 21:00:07 : [INFO ] CURRENT - ABORTED - 0

10/22/2018 21:59:50 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-DPDK-FAILSAFE-DURING-TRAFFIC~~~~~~~~~~
10/22/2018 21:59:50 : [INFO ] CURRENT - PASS    - 1
10/22/2018 21:59:50 : [INFO ] CURRENT - FAIL    - 0
10/22/2018 21:59:50 : [INFO ] CURRENT - ABORTED - 0

10/22/2018 22:20:16 : [INFO ] ~~~~~~~~~~~~~~~TEST END : PERF-DPDK-SINGLE-CORE-PPS-DS4~~~~~~~~~~
10/22/2018 22:20:16 : [INFO ] CURRENT - PASS    - 1
10/22/2018 22:20:16 : [INFO ] CURRENT - FAIL    - 0
10/22/2018 22:20:16 : [INFO ] CURRENT - ABORTED - 0

10/22/2018 22:42:34 : [INFO ] ~~~~~~~~~~~~~~~TEST END : PERF-DPDK-SINGLE-CORE-PPS-DS15~~~~~~~~~~
10/22/2018 22:42:34 : [INFO ] CURRENT - PASS    - 1
10/22/2018 22:42:34 : [INFO ] CURRENT - FAIL    - 0
10/22/2018 22:42:34 : [INFO ] CURRENT - ABORTED - 0

10/22/2018 23:15:01 : [INFO ] ~~~~~~~~~~~~~~~TEST END : PERF-DPDK-MULTICORE-PPS-DS15~~~~~~~~~~
10/22/2018 23:15:01 : [INFO ] CURRENT - PASS    - 1
10/22/2018 23:15:01 : [INFO ] CURRENT - FAIL    - 0
10/22/2018 23:15:01 : [INFO ] CURRENT - ABORTED - 0

Comment (justin-parus) [reply XML/VMConfigurations/PerformanceTestsConfigurations.xml]: All affected and new tests

**Comment by juhlee-microsoft** [comment]
@lizzha @rmsivakanth please review and close your comment if all changes are ok.

**Comment by justin-parus** [comment]
@rmsivakanth Do you have any other suggestions? Otherwise can I merge this?

**Pull request merged**